File: Lab 4/ShapesLibDLL/Shape.cs
using System;
using System.Collections.Generic;
using System.Drawing;


namespace ShapesLibDLL
{

    public struct point
    {
        public int X, Y;

        public int getX()
        {
            return X;
        }
        public int getY()
        {
            return Y;
        }
        public point(int x, int y)
        {
            X = x;
            Y = y;
        }
    };

    public class Shape
    {


        protected List<point> points;

        protected List<double> sides = new List<double>{0};
        public Shape()
        {
        }

        public Shape(List<point> derivedPoints)
        {
            points = derivedPoints;
            
        //   for (int i = 0; i < derivedPoints.Count; i++)
        //   {
        //       points.Add(derivedPoints[i]);
        //   }

            calculateSides();
        }

        public Shape(Shape copy)
        {
            this.points = copy.points;
            this.sides = copy.sides;
        }


        void calculateSides()
        {
            sides.Clear();
            for (int i = 0, j = 1; i < points.Count ; i++, j++)
            {

                if (i == points.Count - 1)
                {
                    sides.Add(Math.Sqrt(Math.Pow(points[0].getX() - points[i].getX(), 2) + Math.Pow(points[0].getY() - points[i].getY(), 2)));
                }
                else
                {
                    sides.Add(Math.Sqrt(Math.Pow(points[i].getX() - points[j].getX(), 2) + Math.Pow(points[i].getY() - points[j].getY(), 2)));
                }
            }
        }

    }
}

File: Lab 4/ShapesLibStatic/shape.cpp
#include "shape.h"

Shape::Shape()
{
}

Shape::Shape(vector<point> derivedPoints)
{	
	points.resize(derivedPoints.size());

	for (int i = 0; i < derivedPoints.size(); i++)
	{
		points[i] = derivedPoints[i];
	}

	calculateSides();
}

Shape::Shape(Shape & copy)
{
	this->points = copy.points;
	this->sides = copy.sides;
}


void Shape::calculateSides()
{
	sides.resize(points.size());

	for (int i = 0, j = 1; i < sides.size() ; i++, j++)
	{
		if (i == sides.size()-1)
		{
			sides[i] = sqrt(pow(points[0].x - points[i].x, 2) + pow(points[0].y - points[i].y, 2));
		}
		else
		{
			sides[i] = sqrt(pow(points[i].x - points[j].x, 2) + pow(points[i].y - points[j].y, 2));
		}
	}
}

